### PR TITLE
Document "ETSConfig" attributes

### LINKED ISSUE #1667: "traits.etsconfig" API documentation can be improved
At the moment, there is a lot of meaningful information on the object that isn't available in the API documentation for https://docs.enthought.com/traits/traits_api_reference/traits.etsconfig.html

Note: It took me a minute to remember that we're manually curating the API documentation so this is where we need to make changes - https://github.com/enthought/traits/blob/66b57b7c1f2127fc070fa943773c840f5f44a9ca/docs/source/traits_api_reference/traits.etsconfig.rst

### PR DESCRIPTION
**Pull request by rahulporuri**
This PR adds documentation about the attributes on `ETSConfig`.

Note to reviewer : I tried adding meaningful descriptions to the various attributes that I could come up with and synthesize from the existing docstrings on the various methods. They still need improvement IMO.

fixes #1667 

**Checklist**
- [ ] ~Tests~
- [ ] ~Update API reference (`docs/source/traits_api_reference`)~
- [ ] ~Update User manual (`docs/source/traits_user_manual`)~
- [ ] ~Update type annotation hints in `traits-stubs`~

### CODE REVIEW

**Comment by mdickinson** [comment]
It would also be good to get @corranwebster's eyes on this, if he has time.

File: traits/etsconfig/etsconfig.py [hunk 1]
```diff
@@ -49,6 +49,26 @@ class ETSConfig(object):
...
+        "application_data" / "package_name".  This directory is meant to store
+        application-specific files e.g. preferences files, log files.
+    company : str, default="Enthought"
+        The company name.
+    toolkit : str
+        GUI toolkit to use.
+
+        If `None`, the `ETS_TOOLKIT` environment variable will be used.
+    kiva_backend : str
+        Kiva backend to use.
```
Comment (corranwebster) [inline traits/etsconfig/etsconfig.py:69]: Might note that this is read-only and dependent on `toolkit`

File: traits/etsconfig/etsconfig.py [hunk 2]
```diff
@@ -49,6 +49,26 @@ class ETSConfig(object):
...
+        non-user accessible data e.g. configuration files.
+    application_home : str
+        Application home directory path. This directory is usually a
+        sub-directory of the application_data directory e.g.
+        "application_data" / "package_name".  This directory is meant to store
+        application-specific files e.g. preferences files, log files.
+    company : str, default="Enthought"
+        The company name.
+    toolkit : str
+        GUI toolkit to use.
```
Comment (corranwebster) [inline traits/etsconfig/etsconfig.py:65]: Might not that this has the form `pyface_toolkit.kiva_backend`, and that if `kiva_backend` is omitted a sensible default is chosen based on OS and GUI toolkit.

File: traits/etsconfig/etsconfig.py [hunk 3]
```diff
@@ -49,6 +49,26 @@ class ETSConfig(object):
...
+
+        If `None`, the `ETS_TOOLKIT` environment variable will be used.
+    kiva_backend : str
+        Kiva backend to use.
+    user_data : str
+        User data directory path.
```
Comment (corranwebster) [inline traits/etsconfig/etsconfig.py:71]: Might add something to the effect of "this is the default location for documents and projects" or something similar.

**Comment by mdickinson** [comment]
I'm not sure that this PR is the right approach: we already have docstrings for the various methods and properties in `ETSConfig`. Those aren't showing up in the built documentation, though, and it would be good to figure out why.

**Comment by mdickinson** [comment]
So maybe the right approach _is_ to rename the `ETSConfig` class, and include it in the API documentation. We can then document the `ETSConfig` instance of that class separately, and include advice about modifying that instance rather than using a new `ETSConfig` instance in tests. (See discussion at #1666.)

**Comment by mdickinson** [comment]
Sorry, @rahulporuri: I'm going to close this in favour of #1688 - I'd like to have a solution where we can maintain the API docs for the individual attributes in a single place.

**Pull request closed without merging**